src/trainer/JasmineGraphTrainingSchedular.cpp

```diff
@@ -0,0 +1,137 @@
+#include "../metadb/SQLiteDBInterface.h"
+#include "../util/Conts.h"
+#include "../frontend/JasmineGraphFrontEnd.h"
+
+
+using namespace std;
+
+void JasmineGraphTrainingSchedular::sortPartions(std::string graphID) {
+
+    vector<pair<string, string>> hostData;
+    SQLiteDBInterface refToSqlite = *new SQLiteDBInterface();
+    refToSqlite.init();
+
+    string sqlStatement =
+            "SELECT idhost, name FROM host_has_partition INNER JOIN host ON host_idhost = idhost WHERE partition_graph_idgraph = '" +
+            to_string(1) + "' group by idhost";
+    std::vector<vector<pair<string, string>>> result = refToSqlite.runSelect(sqlStatement);
+    for (vector<vector<pair<string, string>>>::iterator i = result.begin(); i != result.end(); ++i) {
+        int count = 0;
+        string hostID;
+        string hostname;
+        for (std::vector<pair<string, string>>::iterator j = (i->begin()); j != i->end(); ++j) {
+            if (count == 0) {
+                hostID = j->second;
+            } else {
+                hostname = j->second;
+                hostData.push_back(pair<string, string>(hostID, hostname));
+            }
+            count++;
+        }
+    }
+    string id_partition = "";
+    string vertexCount = "";
+    string centralvertexCount = "";
+    int partition_id;
+    int vertexcount;
+    int centralVertexCount;
+
+    for (std::vector<pair<string, string>>::iterator j = (hostData.begin()); j != hostData.end(); ++j) {
+        sqlStatement =
+                "SELECT idpartition,vertexcount,central_vertexcount,graph_idgraph FROM partition INNER JOIN host_has_partition "
+                "ON partition.idpartition = host_has_partition.partition_idpartition WHERE graph_idgraph = " + graphID +
+                " AND host_idhost = "
+                + j->first;
+        std::vector<vector<pair<string, string>>> results = refToSqlite.runSelect(sqlStatement);
+
+        std::map<int, int> vertexCountToPartitionId;
+        cout << sqlStatement << endl;
+        for (std::vector<vector<pair<string, string>>>::iterator i = results.begin(); i != results.end(); ++i) {
+            std::vector<pair<string, string>> rowData = *i;
+
+            id_partition = rowData.at(0).second;
+            vertexCount = rowData.at(1).second;
+            centralvertexCount = rowData.at(2).second;
+
+            partition_id = stoi(id_partition);
+            vertexcount = stoi(vertexCount);
+            centralVertexCount = stoi(centralvertexCount);
+
+            //ToDO:get triangle count by partition id
+
+            vertexCountToPartitionId.insert({vertexcount + centralVertexCount, partition_id});
+
+
+        }
+
+        float totMemEstimation = 0;
+        int iteration = 0;
+        for (auto i = vertexCountToPartitionId.begin(); i != vertexCountToPartitionId.end(); ++i) {
+            totMemEstimation += estimateMemory(i->first);
+            if (totMemEstimation < 600000000) {  //total=65843492
```

Comment:
We should get numbers such as 600000000 defined as constants in a header file. Also what is the rationale for selecting 600000000 as the value here?

Reply:
This value is taken by chinthaka ayya's work. Dinka has implemented it. she'll replace it here.

Reply:
Ok